woocommerce/wc-admin
# Add wc-api and "Fresh Data"

This adds a wc-api module as a `@wordpress/data` generic store, which is based on [Fresh Data](https://github.com/Automattic/fresh-data)

Note: This PR contains a workaround that will no longer be necessary after the following Gutenberg PR is merged: https://github.com/WordPress/gutenberg/pull/11460

This leverages the general purpose fresh-data framework to create a woocommerce-specific api implementation. The first use of this is in the orders fetching for the orders table in the analytics report. The goal was to have minimal changes to the application code but still gain the advantages of sharing data across components and automatic re-fetching of data.

### Detailed test instructions:

1. `npm install`
6. `npm start`
7. Visit `http://localhost/wp-admin/admin.php?page=wc-admin#/analytics/orders` and ensure that the orders table populates properly at the bottom.
9. Create a new order, and it should show up in the table automatically within 5 minutes.

For additional debugging information, type `localStorage.setItem( 'debug', 'fresh-data:*' );` to see the updating and scheduling of data fetches.
